events_webhook/src/main/java/com/adobe/aio/event/webhook/service/EventVerifier.java

```diff
@@ -78,56 +80,46 @@ public boolean authenticateEvent(String message, String clientId,
     return true;
   }
 
-  private boolean verifyEventSignatures(String message,
-      Map<String, String> headers) {
+  private boolean verifyEventSignatures(String message, Map<String, String> headers) {
     String[] digitalSignatures = {headers.get(ADOBE_IOEVENTS_DIGI_SIGN_1),
         headers.get(ADOBE_IOEVENTS_DIGI_SIGN_2)};
     String[] pubKeyPaths = {headers.get(ADOBE_IOEVENTS_PUB_KEY_1_PATH),
         headers.get(ADOBE_IOEVENTS_PUB_KEY_2_PATH)};
-    String publicKey1Url = ADOBE_IOEVENTS_SECURITY_DOMAIN + headers.get(ADOBE_IOEVENTS_PUB_KEY_1_PATH);
-    String publicKey2Url = ADOBE_IOEVENTS_SECURITY_DOMAIN + headers.get(ADOBE_IOEVENTS_PUB_KEY_2_PATH);
-
-    try {
-      if (isValidUrl(publicKey1Url) && isValidUrl(publicKey2Url)) {
-        return verifySignature(message, pubKeyPaths, digitalSignatures);
-      }
-    } catch (Exception e) {
-      throw new AIOException("Error verifying signature for public keys " + publicKey1Url +
-          " & " + publicKey2Url + ". Reason -> " + e.getMessage());
-    }
-    return false;
+    return verifySignature(message, pubKeyPaths, digitalSignatures);
   }
 
-  private boolean verifySignature(String message, String[] publicKeyPaths, String[] signatures)
-      throws NoSuchAlgorithmException, InvalidKeySpecException, InvalidKeyException, SignatureException {
+  private boolean verifySignature(String message, String[] publicKeyPaths, String[] signatures) {
     byte[] data = message.getBytes(UTF_8);
 
     for (int i = 0; i < signatures.length; i++) {
-      // signature generated at I/O Events side is Base64 encoded, so it must be decoded
-      byte[] sign = Base64.decodeBase64(signatures[i]);
-      Signature sig = Signature.getInstance("SHA256withRSA");
-      sig.initVerify(getPublic(fetchPemEncodedPublicKey(publicKeyPaths[i])));
-      sig.update(data);
-      boolean result = sig.verify(sign);
-      if (result) {
-        return true;
+      try {
+        // signature generated at I/O Events side is Base64 encoded, so it must be decoded
+        byte[] sign = Base64.getDecoder().decode(signatures[i]);
+        Signature sig = Signature.getInstance("SHA256withRSA");
+        sig.initVerify(getPublic(fetchPemEncodedPublicKey(publicKeyPaths[i])));
+        sig.update(data);
+        boolean isSignValid = sig.verify(sign);
```

Comment:
this way is required as we want to check for the next signature if the first signature is invalid

Reply:
I would rather see a simpler and more re-usable public static method such as the one documented here https://developer.adobe.com/events/docs/guides/#security-considerations
with a more straightforward and predictable behavior (that does not assume any cache and does one header verification only ) something like
`verifySignature(String message, String publicKeyPath, String signature)`